### PR TITLE
Fix leaking mach ports & memory after task_threads()

### PR DESCRIPTION
**Pull request by k15tfu**

### CODE REVIEW

**Comment by k15tfu** [comment]
cc @janvorli

File: src/coreclr/pal/src/exception/machmessage.cpp [hunk 1]
```diff
@@ -852,6 +852,11 @@ thread_act_t MachMessage::GetThreadFromState(thread_state_flavor_t eFlavor, thre
                 CHECK_MACH("mach_port_mod_refs()", machret);
```
Comment (k15tfu) [inline src/coreclr/pal/src/exception/machmessage.cpp:852]: I can probably skip deallocation for `pThreads[i]` and remove this manual increment, if needed.

**Comment by k15tfu** [comment]
Friendly ping.

File: src/coreclr/pal/src/exception/machmessage.cpp (hunk 1 above)
Comment (janvorli) [reply src/coreclr/pal/src/exception/machmessage.cpp:852]: I actually prefer the current form.

**Review by janvorli — APPROVED** [review]
LGTM, thank you! I wasn't aware of the fact that the ports need to be deallocated. I've also haven't found it documented anywhere, but test code in the xnu tree does that, so I assume it needs to be done.

**Comment by k15tfu** [comment]
@janvorli Hi! Do you need anything from me to merge it?

**Pull request merged**